googleapi/googleapi.go

```diff
@@ -181,6 +180,22 @@ func CheckResponseWithBody(res *http.Response, body []byte) error {
 	}
 }
 
+// errorReplyFromBody attempts to get the error from body. The body
+// may be a JSON object or JSON array, or may be something else.
+func errorReplyFromBody(body []byte) (*errorReply, error) {
+	jerr := new(errorReply)
+	peekChar := body[0]
+	if peekChar == '[' {
+		// Attempt JSON array
+		jsonArr := &[]*errorReply{jerr}
+		err := json.Unmarshal(body, jsonArr)
```

Comment:
Nit: feels cleaner to let the variable be the value, and let the usage use the reference, at least in the case of a slice - just my two cents
```suggestion
		jsonArr := []*errorReply{jerr}
		err := json.Unmarshal(body, &jsonArr)
```

Reply:
Fixed, thanks!